File: java-client/src/main/java/com/bolour/boardgame/client/domain/StopInfo.java
/*
 * Copyright 2017-2018 Azad Bolour
 * Licensed under GNU Affero General Public License v3.0 -
 *   https://github.com/azadbolour/boardgame/blob/master/LICENSE.md
 */

package com.bolour.boardgame.client.domain;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.Collections;
import java.util.List;

class StopInfo {
    int successivePasses;
    boolean filledBoard;

    @JsonCreator
    public StopInfo(
      @JsonProperty("successivePasses") int successivePasses,
      @JsonProperty("filledBoard") boolean filledBoard
    ) {
        this.successivePasses = successivePasses;
        this.filledBoard = filledBoard;
    }
}

File: scala-server/boardgamecommon/src/main/scala/com/bolour/boardgame/scala/common/domain/StopInfo.scala
/*
 * Copyright 2017-2018 Azad Bolour
 * Licensed under GNU Affero General Public License v3.0 -
 *    https://github.com/azadbolour/boardgame/blob/master/LICENSE.md
 */
package com.bolour.boardgame.scala.common.domain

/**
  * Information about a completed game.
  *
  * TODO. Use sealed abstract case class for different reasons the game stopped.
  *
  * @param successivePasses Number of successive passes (was maxed out).
  * @param filledBoard The board has been filled and no more play is possible.
  */
case class StopInfo(
  successivePasses: Int,
  filledBoard: Boolean
)
